### PR TITLE
TSV export, only expose .json files, create our own index for directory

### PR DESCRIPTION
**Pull request by ldemailly**
This should let fortio servers be programmatically backed up to cloud storage

### CODE REVIEW

**Review by rkpagadala — APPROVED** [review]

**Comment by ldemailly** [comment]
Thanks ! I'm waiting to get it to work before merging (b/70993367)

File: ui/uihandler.go [hunk 1]
```diff
@@ -324,31 +328,35 @@ func SaveJSON(name string, json []byte) string {
 	return "data/" + name
 }
 
-// BrowseHandler handles listing and rendering the JSON results.
-func BrowseHandler(w http.ResponseWriter, r *http.Request) {
-	LogRequest(r, "Browse")
-	url := r.FormValue("url")
-	doRender := (url != "")
+// GetDataList returns the .json files/entries in data dir.
+func GetDataList() (dataList []string) {
```
Comment (douglas-reid) [inline ui/uihandler.go]: super nit: just `DataList()`.

File: ui/uihandler.go [hunk 2]
```diff
@@ -386,10 +394,99 @@ func LogAndAddCacheControl(h http.Handler) http.Handler {
...
+}
+
+var (
+	gTsvCache      tsvCache
+	gTsvCacheMutex = &sync.Mutex{}
+)
+
+// format for gcloud transfer
+// https://cloud.google.com/storage/transfer/create-url-list
+func sendTsvDataIndex(urlPrefix string, w http.ResponseWriter) {
```
Comment (douglas-reid) [inline ui/uihandler.go]: should this be: `sendTSVDataIndex` ?

I think TSV should be all-caps everywhere.

**Review by douglas-reid — COMMENTED** [review]
/lgtm

only style nits.

**Review by douglas-reid — APPROVED** [review]

**Comment by ldemailly** [comment]
going to merge it because of the `report` mode fixes. even though it doesn't yet quite work on the cloud side and will make follow up PRs as needed as this is reviewed already

**Pull request merged**